update: follower fetching content updated to include GCP (#378)
aiven/aiven-docs: docs/products/kafka/howto/enable-follower-fetching.md

- [Aiven Provider for Terraform](https://registry.terraform.io/providers/aiven/aiven/latest/docs).

:::note
Follower fetching is supported on AWS (Amazon Web Services) and

> GCP (Google Cloud Platform).

Reviewer: Same as other comment

:::

## Identify availability zone

- **AWS**: In AWS, availability zone (AZ) names can vary across different accounts.
  The same physical location might have different AZ names in different accounts. To
  ensure consistency when configuring `client.rack`, use the AZ ID, which remains the same
  across accounts.

>   To map AZ names to AZ IDs, see

Reviewer: ```suggestion
  To map AZ names to AZ IDs, see the
```

  [AWS Knowledge Center article](https://repost.aws/knowledge-center/vpc-map-cross-account-availability-zones)
  and the [AWS documentation on AZ IDs](https://docs.aws.amazon.com/ram/latest/userguide/working-with-az-ids.html).

> - **GCP**: In GCP, use the AZ name directly as the `client.rack` value.

Reviewer: Same as other comments on GCP

## Client-side configuration

To enable follower fetching at the client level, configure the `client.rack` setting
in the Apache Kafka client. Set the `client.rack` value to the corresponding AZ ID for

> AWS or AZ name for GCP for each client. This ensures the client fetches data from the

Reviewer: Same for GCP here

nearest replica.

Example configuration for your consumer properties file:

```plaintext
client.rack=use1-az1 # AWS example

> client.rack=europe-west1-b # GCP example

Reviewer: ```suggestion
client.rack=europe-west1-b # Google Cloud example
```

```

### Example scenario: follower fetching in different AZs

Assume you have an Aiven for Apache Kafka cluster running in two AZs in the `us-east-1`

> region for AWS and in the `europe-west1` region for GCP:

Reviewer: ```suggestion
region for AWS and in the `europe-west1` region for Google Cloud:
```

#### Cluster setup and consumer distribution

| Cloud | Region         | AZs for brokers               | AZs for consumers                             |
|-------|----------------|-------------------------------|-----------------------------------------------|
| AWS   | `us-east-1`    | `use1-az1`, `use1-az2`        | `use1-az1`, `use1-az2`, `use1-az3`            |

> | GCP   | `europe-west1` | `europe-west1-b`, `europe-west1-c` | `europe-west1-b`, `europe-west1-c`, `europe-west1-d` |

Reviewer: ```suggestion
| Google Cloud   | `europe-west1` | `europe-west1-b`, `europe-west1-c` | `europe-west1-b`, `europe-west1-c`, `europe-west1-d` |
```

#### Consumer configuration

> Set the `client.rack` value to the respective AZ ID for AWS or AZ name for GCP for each consumer:

Reviewer: ```suggestion
Set the `client.rack` value to the respective AZ ID for AWS or AZ name for Google Cloud for each consumer:
```

```plaintext
# AWS consumers in use1-az1
client.rack=use1-az1

# AWS consumers in use1-az2
client.rack=use1-az2

# AWS consumers in use1-az3
client.rack=use1-az3

> # GCP consumers in europe-west1-b

Reviewer: Same for all of these
